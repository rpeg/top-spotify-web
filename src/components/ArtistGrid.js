import React from 'react';
import { Row, Col } from 'react-bootstrap';
import { chunk, isEmpty } from 'lodash';

const ArtistGrid = ({ artists }) => {
  const chunkedArtists = chunk(artists, 5);

  return chunkedArtists.map((artistRow, i) => (
    <div key={artistRow.map((a) => a.id).toString()}>
      <Row key={chunkedArtists[i].map((a) => a.id).toString()} style={{ paddingTop: `${i > 0 ? '4px' : '0px'}` }}>
        {artistRow.map((artist) => (
          <Col key={artist.id} style={{ padding: '0px 2px 0px 2px' }}>
            <div style={{
              display: 'flex', paddingBottom: '100%', overflow: 'hidden', position: 'relative',
            }}
            >
              <img
                src={!isEmpty(artist.images) && artist.images[0].url ? artist.images[0].url : 'images/person.svg'}
                alt={artist.name}
                style={{
                  width: '100%',
                  height: '100%',
                  position: 'absolute',
                  objectFit: 'cover',
                  objectPosition: 'center',
                }}
              />
              <div
                style={{
                  position: 'absolute',
                  zIndex: '10',
                  width: '100%',
                  bottom: '0',
                  marginBottom: '0',
                  background: 'rgba(0,0,0,0.5)',
                }}
              >
                <p
                  className="s"
                  style={{
                    marginBottom: '0',
                    color: 'white',
                  }}
                >
                  <b>{artist.name}</b>
                </p>
              </div>
            </div>
          </Col>
        ))}
      </Row>
    </div>
  ));
};

export default ArtistGrid;
