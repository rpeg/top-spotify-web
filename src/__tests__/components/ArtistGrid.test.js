import React from 'react';
import { shallow } from 'enzyme';

import ArtistGrid from '../../components/ArtistGrid';

describe('ArtistGrid', () => {
  it('should not log errors in console', () => {
    const wrapper = shallow(<ArtistGrid />);
    const spy = jest.spyOn(global.console, 'error');
    expect(wrapper).not.toBeNull();
    expect(spy).not.toHaveBeenCalled();
  });

  it('does not render without artists', () => {
    const wrapper = shallow(<ArtistGrid />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders with artists', () => {
    const artists = [
      {
        id: 'abc',
        name: 'A',
      },
      {
        id: 'cde',
        name: 'B',
      },
      {
        id: 'fgh',
        name: 'C',
      },
      {
        id: 'ijk',
        name: 'D',
      },
      {
        id: 'lmn',
        name: 'E',
      },
    ];

    const wrapper = shallow(<ArtistGrid artists={artists} />);
    expect(wrapper).toMatchSnapshot();
  });
});
