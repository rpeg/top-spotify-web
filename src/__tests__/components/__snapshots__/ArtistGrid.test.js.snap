// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistGrid does not render without artists 1`] = `null`;

exports[`ArtistGrid renders with artists 1`] = `
<div
  key="abc,cde,fgh,ijk,lmn"
>
  <ForwardRef
    key="abc,cde,fgh,ijk,lmn"
    noGutters={false}
    style={
      Object {
        "paddingTop": "0px",
      }
    }
  >
    <Col
      key="abc"
      style={
        Object {
          "padding": "0px 2px 0px 2px",
        }
      }
    >
      <div
        style={
          Object {
            "display": "flex",
            "overflow": "hidden",
            "paddingBottom": "100%",
            "position": "relative",
          }
        }
      >
        <img
          alt="A"
          src="images/person.svg"
          style={
            Object {
              "height": "100%",
              "objectFit": "cover",
              "objectPosition": "center",
              "position": "absolute",
              "width": "100%",
            }
          }
        />
        <div
          style={
            Object {
              "background": "rgba(0,0,0,0.5)",
              "bottom": "0",
              "marginBottom": "0",
              "position": "absolute",
              "width": "100%",
              "zIndex": "10",
            }
          }
        >
          <p
            className="s"
            style={
              Object {
                "color": "white",
                "marginBottom": "0",
              }
            }
          >
            <b>
              A
            </b>
          </p>
        </div>
      </div>
    </Col>
    <Col
      key="cde"
      style={
        Object {
          "padding": "0px 2px 0px 2px",
        }
      }
    >
      <div
        style={
          Object {
            "display": "flex",
            "overflow": "hidden",
            "paddingBottom": "100%",
            "position": "relative",
          }
        }
      >
        <img
          alt="B"
          src="images/person.svg"
          style={
            Object {
              "height": "100%",
              "objectFit": "cover",
              "objectPosition": "center",
              "position": "absolute",
              "width": "100%",
            }
          }
        />
        <div
          style={
            Object {
              "background": "rgba(0,0,0,0.5)",
              "bottom": "0",
              "marginBottom": "0",
              "position": "absolute",
              "width": "100%",
              "zIndex": "10",
            }
          }
        >
          <p
            className="s"
            style={
              Object {
                "color": "white",
                "marginBottom": "0",
              }
            }
          >
            <b>
              B
            </b>
          </p>
        </div>
      </div>
    </Col>
    <Col
      key="fgh"
      style={
        Object {
          "padding": "0px 2px 0px 2px",
        }
      }
    >
      <div
        style={
          Object {
            "display": "flex",
            "overflow": "hidden",
            "paddingBottom": "100%",
            "position": "relative",
          }
        }
      >
        <img
          alt="C"
          src="images/person.svg"
          style={
            Object {
              "height": "100%",
              "objectFit": "cover",
              "objectPosition": "center",
              "position": "absolute",
              "width": "100%",
            }
          }
        />
        <div
          style={
            Object {
              "background": "rgba(0,0,0,0.5)",
              "bottom": "0",
              "marginBottom": "0",
              "position": "absolute",
              "width": "100%",
              "zIndex": "10",
            }
          }
        >
          <p
            className="s"
            style={
              Object {
                "color": "white",
                "marginBottom": "0",
              }
            }
          >
            <b>
              C
            </b>
          </p>
        </div>
      </div>
    </Col>
    <Col
      key="ijk"
      style={
        Object {
          "padding": "0px 2px 0px 2px",
        }
      }
    >
      <div
        style={
          Object {
            "display": "flex",
            "overflow": "hidden",
            "paddingBottom": "100%",
            "position": "relative",
          }
        }
      >
        <img
          alt="D"
          src="images/person.svg"
          style={
            Object {
              "height": "100%",
              "objectFit": "cover",
              "objectPosition": "center",
              "position": "absolute",
              "width": "100%",
            }
          }
        />
        <div
          style={
            Object {
              "background": "rgba(0,0,0,0.5)",
              "bottom": "0",
              "marginBottom": "0",
              "position": "absolute",
              "width": "100%",
              "zIndex": "10",
            }
          }
        >
          <p
            className="s"
            style={
              Object {
                "color": "white",
                "marginBottom": "0",
              }
            }
          >
            <b>
              D
            </b>
          </p>
        </div>
      </div>
    </Col>
    <Col
      key="lmn"
      style={
        Object {
          "padding": "0px 2px 0px 2px",
        }
      }
    >
      <div
        style={
          Object {
            "display": "flex",
            "overflow": "hidden",
            "paddingBottom": "100%",
            "position": "relative",
          }
        }
      >
        <img
          alt="E"
          src="images/person.svg"
          style={
            Object {
              "height": "100%",
              "objectFit": "cover",
              "objectPosition": "center",
              "position": "absolute",
              "width": "100%",
            }
          }
        />
        <div
          style={
            Object {
              "background": "rgba(0,0,0,0.5)",
              "bottom": "0",
              "marginBottom": "0",
              "position": "absolute",
              "width": "100%",
              "zIndex": "10",
            }
          }
        >
          <p
            className="s"
            style={
              Object {
                "color": "white",
                "marginBottom": "0",
              }
            }
          >
            <b>
              E
            </b>
          </p>
        </div>
      </div>
    </Col>
  </ForwardRef>
</div>
`;
